include/swift/AST/LazyResolver.h

```diff
@@ -218,12 +218,23 @@ class alignas(void*) LazyMemberLoader {
 public:
   virtual ~LazyMemberLoader() = default;
 
-  /// Populates the given vector with all member decls for \p D.
+  /// Populates a given decl \p D with all of its members.
   ///
   /// The implementation should add the members to D.
   virtual void
   loadAllMembers(Decl *D, uint64_t contextData) = 0;
 
+  /// Populates a vector with all members of \p D that have DeclName
+  /// matching \p N.
+  ///
+  /// Returns true if an error occurred \em or the set of returned
+  /// \p Members is otherwise incomplete, due to implementation limitations
+  /// (eg. the implementation is unable to do named lookup at all, or within
+  /// the particular type of Decl that \p D is).
+  virtual bool
```

Comment:
Would this API be cleaner if, instead of a `bool` result type and by-reference vector, it returned an `Optional<TinyPtrVector<ValueDecl *>>` ?

Reply:
I guess it means you can't get an "incomplete" result, where `Members` is partially filled but can't be completely filled, but that seems okay?

Reply:
Happy to change it to Optional, don't feel strongly. You prefer Optional here?